### DISCUSSION TITLE
setup oidc gitlab

### DISCUSSION THREAD

**Discussion by bourribab** [DependencyTrack/dependency-track#1377] [General]
hello,

I am new to owasp dependency track and I want to configure oidc with gitlab .
 my configuration apifront:

OIDC_ISSUER=http:*******"
OIDC_CLIENT_ID=***********

My configuration  apiserver for enable oidc with gitlab : 

       - "ALPINE_OIDC_ENABLED=true"
       - "ALPINE_OIDC_CLIENT_ID=f********************"
       - "ALPINE_OIDC_ISSUER=http:/***********"
       - "ALPINE_OIDC_USERNAME_CLAIM=nickname"
       - "ALPINE_OIDC_TEAMS_CLAIM=groups"
       - "ALPINE_OIDC_USER_PROVISIONING=true"
       - "ALPINE_OIDC_TEAM_SYNCHRONIZATION=true"
    # Optional Cross-Origin Resource Sharing (CORS) Headers
       - "ALPINE_CORS_ENABLED=true"
       - "ALPINE_CORS_ALLOW_ORIGIN=*"
       - "ALPINE_CORS_ALLOW_METHODS=GET POST PUT DELETE OPTIONS"
       - "ALPINE_CORS_ALLOW_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *"
       - "ALPINE_CORS_EXPOSE_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count"
       - "ALPINE_CORS_ALLOW_CREDENTIALS=true"
       - "ALPINE_CORS_MAX_AGE=3600"

I have this error on my browser:

Access to XMLHttpRequest at 'http://********/.well-known/openid-configuration' from origin 'http://localhost:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

do you have an idea please?

**Comment by nscuro** [comment]
Hi @bourribab, please ensure that you're using http**S** URLs for GitLab. They *do* set the required headers AFAICT:

```shell
$ curl --head -H 'Origin: example.com' https://gitlab.com/.well-known/openid-configuration
HTTP/2 200
date: Tue, 08 Feb 2022 11:04:51 GMT
content-type: application/json; charset=utf-8
access-control-allow-methods: GET, HEAD
access-control-allow-origin: *
access-control-expose-headers:
access-control-max-age: 7200
...
```

However, when using http, you'll get a redirect to https, and the CORS headers are not set:

```shell
$ curl --head -H 'Origin: example.com' http://gitlab.com/.well-known/openid-configuration
HTTP/1.1 301 Moved Permanently
Date: Tue, 08 Feb 2022 11:05:19 GMT
Connection: keep-alive
Cache-Control: max-age=3600
Expires: Tue, 08 Feb 2022 12:05:19 GMT
Location: https://gitlab.com/.well-known/openid-configuration
...
```

When using a self-hosted GitLab instance, make sure you're running at least version 14.0. That seems to be the version in which CORS headers have been added to the OIDC endpoints (see https://gitlab.com/gitlab-org/gitlab/-/issues/209259).

**Reply by bourribab** [reply]
Thank you for your feedback. I will test your solution and I will leave you a message.

**Reply by bourribab** [reply]
the problem is solved by using https.
